Code/Framework/AzCore/AzCore/Asset/AssetCommon.h

```diff
@@ -556,16 +556,24 @@ namespace AZ
                     Asset<AssetData> assetData(AssetInternal::GetAssetData(actualId, AZ::Data::AssetLoadBehavior::Default));
                     if (assetData)
                     {
-                        auto curStatus = assetData->GetStatus();
+                        auto isReady = assetData->GetStatus() == AssetData::AssetStatus::Ready;
                         bool isError = assetData->IsError();
-                        connectLock.unlock();
-                        if (curStatus == AssetData::AssetStatus::Ready)
-                        {
-                            handler->OnAssetReady(assetData);
-                        }
-                        else if (isError)
+
+                        if (isReady || isError)
                         {
-                            handler->OnAssetError(assetData);
+                            connectLock.unlock();
```

Comment:
So how do you know if the thread actually had the lock?

Reply:
Hm, you've highlighted the .unlock line, but it kinda sounds like you're asking "if the thread didn't have the lock in the first place, shouldn't we avoid calling .lock".  The .unlock was already there, so I would assume that's not an issue.  Also I'm not sure how a thread wouldn't have the lock, it seems all the calling code locks the mutex before it gets here.

Reply:
Yes the EBus has the locked the context mutex when the [Connect](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/EBus/EBus.h#L1004-L1027) function is called.